### PR TITLE
🐛 Fixes ``/scheduled_maintenance`` entrypoint

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- 🐛 Protects ``/v0/scheduled_maintenance`` entrypoint
   - @odeimaiz do we need modifications in the front-end? i.e. do do this request after login?
- ♻️ adds tests to entrypoint
- EXTRA: ♻️ 📝 doc and annotations of ``socketio``

## How to test

- ``cd services/web/server``
- ``make install-dev``
- ``pytest -vvtests/unit/**/test_rest.py``

### CODE REVIEW

**Comment by odeimaiz** [comment]
@pcrespov All good

File: services/web/server/src/simcore_service_webserver/login/decorators.py [hunk 1]
```diff
@@ -16,6 +16,8 @@ def login_required(handler):
     @wraps(handler)
     async def wrapped(*args, **kwargs):
         request = get_request(*args, **kwargs)
+        # WARNING: not that check_authorized is patched in some tests.
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/login/decorators.py:19]:
```suggestion
        # WARNING: note that check_authorized is patched in some tests.
```

File: services/web/server/src/simcore_service_webserver/socketio/handlers_utils.py [hunk 2]
```diff
@@ -1,13 +1,35 @@
...
+]
+
+# Disconnect event
+SocketioDisconnectEventHandler = Callable[[SocketID, web.Application], Awaitable[None]]
+
+# Event
+AnyData = Any
+SocketioEventHandler = Callable[[SocketID, AnyData, web.Application], Awaitable[None]]
+
+_socketio_handlers_registry: list[
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/socketio/handlers_utils.py:26]: Q: should we not remove that from the global scope?

File: services/web/server/src/simcore_service_webserver/rest_handlers.py [hunk 3]
```diff
@@ -82,13 +83,18 @@ async def get_config(request: web.Request):
...
-    # {"start": "2023-01-17T14:45:00.000Z", "end": "2023-01-17T23:00:00.000Z", "reason": "Release 1.0.4"}
+    # Examples.
+    #  {"start": "2023-01-17T14:45:00.000Z", "end": "2023-01-17T23:00:00.000Z", "reason": "Release 1.0.4"}
+    #  {"start": "2023-01-20T09:00:00.000Z", "end": "2023-01-20T10:30:00.000Z", "reason": "Release ResistanceIsFutile2"}
+    # NOTE: datetime is UTC (Canary islands / UK)
     if maintenance_str := await redis_client.get(hash_key):
         return web.json_response(data={"data": maintenance_str})
 
-    return web.json_response(status=204)
+    assert web.HTTPNoContent.status_code == 204  # nosec
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/rest_handlers.py:99]: this is a bit weird...

**Review by sanderegg — APPROVED** [review]

**Review by GitHK — APPROVED** [review]

**Review by mrnicegyu11 — APPROVED** [review]
thanks!

File: services/web/server/src/simcore_service_webserver/socketio/handlers_utils.py (hunk 2 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/socketio/handlers_utils.py:26]: yes, I agree. I prefer using app scope perhaps but in a different PR (this is some cleanup i did for https://github.com/ITISFoundation/osparc-simcore/issues/3387)

File: services/web/server/src/simcore_service_webserver/rest_handlers.py (hunk 3 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/rest_handlers.py:99]: haha, well the story goes as follows. I see
```python
return web.json_response(status=204)
```
but, what was 204 ? aha, no content
```python
return web.json_response(status=web.HTTPNoContent.status_code)
```
wait, what was the status code again?
```python
return web.json_response(status=web.HTTPNoContent) # 204
```
mmhh there is a risk the comment does not get change when the status code changes
```python
assert web.HTTPNoContent.status_code == 204  # nosec
return web.json_response(status=web.HTTPNoContent)
```
well, let's connect both
```python
    response = web.json_response(status=web.HTTPNoContent.status_code)
    assert response.status == 204  # nosec
    return response
```

Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/rest_handlers.py:99]: starlette.status solves this dilemma nicely by including both the status name and code in enums names!

**Review by matusdrobuliak66 — APPROVED** [review]

**Pull request merged**